Add open-graph tags and logo

ashwini0529 opened the pull request:
Open-Graph tags added helps while sharing link on facebook.
Closes https://github.com/coala/website/issues/68

ashwini0529 commented:
I was wondering if someone can review my PR.

Nosferatul commented on _posts/2000-01-01-intro.md:
```diff
@@ -5,6 +5,8 @@ color: black
 style: center
 ---
 
+<img src="{{ site.logo}} " width="150px" height="150px">
```
The logo isn't appropriate, I see a white box.

ashwini0529 replied:
I took the logo from `api.coala.io`
Can you please tell me the link from where should I take the logo?

sils replied:
from the coala-artwork repository

Mariatta replied:
Is it possible to use css styling instead of inline styles ?

eg move width and height into css.

also spacing issue
should be:
`"{{ site.logo }}"`
instead of `"{{ site.logo}} "`

ashwini0529 replied:
Sure. I'll update my PR.

Mariatta replied:
I realized it's not possible lol
Ignore me.

But spacing issue

ashwini0529 replied:
@Mariatta done. :smile: 
Everything is possible in code! :smile:

Nosferatul commented:
Be sure that we will review your pr, you can find another issue to work on 😄

sils commented:
@ashwini0529 can you post a screenshot for such changes?

ashwini0529 commented:
@sils It looks something like this.
![selection_044](https://cloud.githubusercontent.com/assets/8419333/21272138/e877ca04-c3e3-11e6-9303-feb6c6d9ac7c.png)

Mariatta commented on _posts/2000-01-01-intro.md:
```diff
@@ -5,6 +5,8 @@ color: black
 style: center
 ---
 
+<img class = "coala-logo" src="{{ site.logo }}">
```
Awesome!! :tada:

but 
`class="coala-logo"`

sorry I'm so nitpicky :( and pretty much a human linter :sweat:

ashwini0529 replied:
So what exactly do you want the class name to be?😅
Or do you want the space to be removed?

Mariatta replied:
the space to be removed

ashwini0529 replied:
Alright. Sorry for such mistakes 😅

ashwini0529 replied:
@Mariatta Done! Good to go? :smile:

Mariatta commented:
LGTM Thanks :)

Mariatta commented:
Squash commits

ashwini0529 commented:
@Mariatta but both the commits serve different purpose. That's why two different commits.
Adding `open graph` tags and adding `logo` are completely different tasks and closes two different issues.
Squashing it into one commit would create confusion in this case.
I guess two commits would work

Mariatta commented:
Then maybe you need two separate PRs cc @sils for opinion

ashwini0529 commented:
Well a single PR can bring many changes that can close multiple issues. Creating two PRs will not be apt.
IMO One issue can't have many PRs, but one PR can close many issues.

jayvdb commented on _includes/css/main.css:
```diff
@@ -355,3 +355,9 @@ nav ul li:hover, nav ul li.active {
+.coala-logo {
+  height: 150px;
+  width: 150px;
+}
```
Missing EOL here

ashwini0529 replied:
Will do it

ashwini0529 replied:
Done!

jayvdb commented on _config.yml:
```diff
@@ -19,8 +19,8 @@ keywords:    "coala, review, analysis, static, bears"
 favicon:     "https://raw.githubusercontent.com/coala-analyzer/coala-artwork/master/artwork/coala_bear.png"
-logo:        "http://api.coala.io/en/latest/_static//images/coala_logo.svg"
-
+logo:        "img/coala_logo.png"
```
Why is it necessary to switch from svg to png?

ashwini0529 replied:
It had some white background.  @Nosferatul suggested to use other picture.

jayvdb replied:
The SVG has a white background?  So, can the SVG be fixed to have a transparent background?

ashwini0529 replied:
Well I can try doing that.

sils commented on _includes/css/main.css:
```diff
@@ -355,3 +355,9 @@ nav ul li:hover, nav ul li.active {
   }
 }
+
+/*Coala logo height and width */
```
coala is always written with a lower case c

ashwini0529 replied:
Haha yeah my bad!

ashwini0529 replied:
Done!

jayvdb commented on index.html:
```diff
@@ -9,6 +9,11 @@
   <title>{{ site.title }}</title>
   <meta name="keywords" content="{{ site.keywords }}">
   <meta name="description" content="{{ site.description }}">
+  <meta property="og:url" content="https://coala.io/
```
Remove the `\n` from the URL.

jayvdb commented:
And please squash your commits.  Your second commit now modifies the same lines that were modified in your first commit, which is a good indication that you shouldnt have two commits.

ashwini0529 commented:
Sorry for the delay. Will make changes in a while.

jayvdb commented on _config.yml:
```diff
@@ -19,7 +19,8 @@ keywords:    "coala, review, analysis, static, bears"
 source_link: "https://github.com/coala-analyzer/coala/"
 favicon:     "https://raw.githubusercontent.com/coala-analyzer/coala-artwork/master/artwork/coala_bear.png"
-
+logo:        "img/coala_logo.svg"
```
why cant we use https://github.com/coala/artwork/blob/master/artwork/logo/coala_color.svg
i.e. https://raw.githubusercontent.com/coala/artwork/master/artwork/logo/coala_color.svg 
it has a mime-type of `text/plain` . is that the problem.

ashwini0529 replied:
We've used this one only. I've just renamed it and saved it in our project directory.

ashwini0529 replied:
This is the same file only. I've just downloaded the logo and used it over here.

jayvdb replied:
But , why cant we use the original in the artwork repo?  Copying will mean this copy becomes out of sync if the origin artwork is modified.

ashwini0529 replied:
Alright. I'll use the link of artwork repo. i.e. https://raw.githubusercontent.com/coala/artwork/master/artwork/logo/coala_color.svg

ashwini0529 replied:
@jayvdb I tried doing that, but the image is not loading due to type.
We need to change `htaccess` I guess and add `svg=image/svg+xml` .
For the time being, is this good to go or you have a better option?

jayvdb commented:
Please squash your commits first, before doing anything else.

ashwini0529 commented:
@jayvdb Yeah sure. Sorry for the delay.

Nosferatul commented:
You have some conflicts, solve them

ashwini0529 committed:
Add open graph tags and logo

Open-Graph tags added helps while sharing link on facebook.
Logo and favicon added.
Closes https://github.com/coala/website/issues/68, https://github.com/coala/website/issues/39

ashwini0529 commented:
@jayvdb Done! Is this good to go now?

ashwini0529 commented:
@Nosferatul No conflicts now :smile:

jayvdb commented:
Waiting on a reply to https://github.com/coala/website/pull/69/files#r93688162

ashwini0529 commented on _config.yml:
```diff
@@ -19,7 +19,8 @@ keywords:    "coala, review, analysis, static, bears"
 source_link: "https://github.com/coala-analyzer/coala/"
 favicon:     "https://raw.githubusercontent.com/coala-analyzer/coala-artwork/master/artwork/coala_bear.png"
-
+logo:        "https://rawgit.com/coala/artwork/master/artwork/logo/coala_color.svg"
```
@jayvdb I've fixed the issue. http://rawgit.com/ is a website which is used to serve `svgs`.

jayvdb commented:
ack f88fd3b

jayvdb commented:
@rultor merge

rultor commented:
> @rultor merge

@jayvdb OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/10392-269072109)

rultor commented:
> @rultor merge

@ashwini0529 @jayvdb Oops, I failed. You can see the full log [here](http://www.rultor.com/t/10392-269072109) (spent 9s)

```
+ set -e
+ set -o pipefail
++ dirname ./run.sh
+ cd .
+ echo 24847
+ echo '1.62.2 ce7ad22'
1.62.2 ce7ad22
+ date
Sat Dec 24 06:47:46 UTC 2016
+ uptime
 06:47:46 up 35 days, 13:02,  0 users,  load average: 8.96, 7.86, 4.81
+ fork=git@github.com:ashwini0529/website.git
+ fork_branch=open-graphs
+ head=git@github.com:coala/website.git
+ head_branch=gh-pages
+ pull_id=69
+ pull_title='Add open-graph tags and logo'
+ vars=('--env=fork=git@github.com:ashwini0529/website.git' '--env=fork_branch=open-graphs' '--env=head=git@github.com:coala/website.git' '--env=head_branch=gh-pages' '--env=pull_id=69' '--env=pull_title=Add open-graph tags and logo')
+ image=yegor256/rultor
+ directory=
+ scripts=()
+ container=coala_website_69
+ as_root=false
+ mkdir -p /home/rultor/.ssh
+ echo -e 'Host github.com\n\tStrictHostKeyChecking no\n'
+ chmod 600 /home/rultor/.ssh/config
+ git clone git@github.com:coala/website.git repo
Cloning into 'repo'...
+ cd repo
+ git config user.email me@rultor.com
+ git config user.name rultor
+ '[' -z '' ']'
+ '[' -e pom.xml ']'
+ '[' -e Gemfile ']'
+ '[' -e build.xml ']'
+ [[ -e build.sbt ]]
+ [[ -e project/Build.scala ]]
+ echo 'I can'\''t guess your build automation tool, see http://doc.rultor.com/basics.html'
I can't guess your build automation tool, see http://doc.rultor.com/basics.html
+ exit -1
'cid' file is absent, container wasn't started correctly

```

ashwini0529 commented:
I think @rultor is not integrated.

rultor commented:
> I think @rultor is not integrated. 

@ashwini0529 I see you're talking about me, but I don't understand it. If you want to say something to me directly, start a message with @rultor

jayvdb commented:
OK, could you rebase please, and then I can do a ff merge.

ashwini0529 commented:
@jayvdb done! Can be merged now! 😄

jayvdb commented:
done.
